### PR TITLE
Feature/side-panel-open-close

### LINKED ISSUE #192: Implement toolbar to browse main entities
Currently, Systems, FTA trees, FMEA worksheets are all shown on the main screen. The goal of this ticket is to put each of them on a separate screen and create a toolbar on the left to navigate among them as can be shown here:
image

For some inspiration see how it works in [termit-dev](https://kbss.felk.cvut.cz/termit-dev/#/):
image

A/C:
- panel allows to navigate among to separate screens for Systems, FTA trees, FMEA worksheets

### PR DESCRIPTION
**Pull request by Kasmadei**
Resolves: https://github.com/kbss-cvut/fta-fmea-ui/issues/192

Not sure how to place button better or another way.
Screenshot 2024-03-04 at 7 15 34 PM
Screenshot 2024-03-04 at 7 15 41 PM

### CODE REVIEW

**Review by blcham — APPROVED** [review]
LGTM

**Comment by Kasmadei** [comment]
@blcham  Do we need to wait for another review or we can merge it ?

**Comment by blcham** [comment]
You can merge since I am the only reviewer here and I approved it.

But it is a good question to clear as in the [previous PR](https://github.com/akaene/fta-fmea-ui/pull/2) I approved PR with the message "LGTM, good idea to use localStorage. Please wait for @ grotskat to finish the review.". 

Regarding the semantics of "approved PR":
- AFAIK, the "approved PR" in GitHub means no more than "reviewer thinks that it is ok" but does not say in general if you are allowed to merge PR. As [repository can be configured to require one or multiple reviewers to merge PR](https://github.blog/2018-03-23-require-multiple-reviewers/), it is up to the repository rules to allow you / not allow you to merge PR.
- so **in general** the way to decide if you can merge is -- whenever you see "Merge pull request" button enabled you are allowed to hit it and merge :)

On the other hand, I did not configure the repository to strictly force the person to wait for the approval of PR. So, I leave this option to the PR owner, and until it works in our group well, I will most likely leave it like that.
image

Within the  [previous PR](https://github.com/akaene/fta-fmea-ui/pull/2), it would have been better if I did not approve the PR but only commented on it and left @ grotskat to approve.

So, to sum up,  thanks for the raised question; I will rethink the configuration of repositories and whether it is worth reconfiguring repositories while keeping the flexibility. Until then, we have a convention that every PR should be reviewed at least by one person until you decide otherwise, but at least tag me there so that I know about it for sure :). @ledsoft

**Comment by ledsoft** [comment]
@blcham Since you tagged me, here are my two cents how this can be solved:
1. Set up protected branches, requiring PR approval by at least one reviewer for merge. Repository owners can still merge PRs without waiting for approval (that's how TermIt repo is set up).
2. Have developers work in their own fork and only repository owners could merge PRs into the main repository (as we discussed this week).

Otherwise, my understanding always was that if a PR is approved, it could be merged and actually even the reviewer can (or maybe even should) merge it.
